app/services/data_migrations/backfill_degrees_new_data.rb

```diff
@@ -0,0 +1,72 @@
+module DataMigrations
+  class BackfillDegreesNewData
+    TIMESTAMP = 20220307165548
+    MANUAL_RUN = true
+
+    def change
+      ApplicationQualification.degree.find_each do |qualification|
+        degree_type = degree_type_for(qualification)
+        institution = institution_for(qualification)
+        subject = subject_for(qualification)
+        grade = grade_for(qualification)
+
+        if [degree_type, institution, subject, grade].any?
+          qualification.update(
```

Comment:
I think that this, because it runs callbacks, will trigger `application_form.touch_choices` (see `models/concerns/touch_application_choices.rb`). This will

- cause the associated application choices to get an `updated_at` of now, which will cause them to appear in the Vendor API as if changed
- throw exceptions when we try to update qualifications on a app from a previous recruitment cycle (see `ApplicationForm#touch_choices`)

Reply:
Good feedback! I am not in my Mac now but I will enter late to check this. So maybe we can use the [update_columns](https://api.rubyonrails.org/classes/ActiveRecord/Persistence.html#method-i-update_columns), wdyt?

Reply:
Sounds good - unless we also need validations to run?

Reply:
No, we don't need validations to run, the only purpose is to update the UUID columns. That's done ♻️